x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/fielddata/ScriptLongFieldData.java

```diff
@@ -0,0 +1,171 @@
+    }
+
+    private final Index index;
+    private final String fieldName;
+    private final LongScriptFieldScript.Factory scriptFactory;
+    private final SetOnce<LongScriptFieldScript.LeafFactory> leafFactory = new SetOnce<>();
+
+    private ScriptLongFieldData(Index index, String fieldName, LongScriptFieldScript.Factory scriptFactory) {
+        this.index = index;
+        this.fieldName = fieldName;
+        this.scriptFactory = scriptFactory;
+    }
+
+    @Override
+    public void setSearchLookup(SearchLookup searchLookup) {
+        // TODO wire the params from the mappings definition, we don't parse them yet
+        this.leafFactory.set(scriptFactory.newFactory(Collections.emptyMap(), searchLookup));
+    }
+
+    @Override
+    public String getFieldName() {
+        return fieldName;
+    }
+
+    @Override
+    public ValuesSourceType getValuesSourceType() {
+        return CoreValuesSourceType.NUMERIC;
+    }
+
+    @Override
+    public ScriptLongLeafFieldData load(LeafReaderContext context) {
+        try {
+            return loadDirect(context);
+        } catch (Exception e) {
+            if (e instanceof ElasticsearchException) {
+                throw (ElasticsearchException) e;
+            } else {
+                throw new ElasticsearchException(e);
+            }
+        }
+    }
+
+    @Override
+    public ScriptLongLeafFieldData loadDirect(LeafReaderContext context) throws IOException {
+        return new ScriptLongLeafFieldData(new ScriptLongDocValues(leafFactory.get().newInstance(context)));
+    }
+
+    @Override
+    public NumericType getNumericType() {
+        return NumericType.LONG;
+    }
+
+    @Override
+    protected boolean sortRequiresCustomComparator() {
+        return true;
+    }
+
+    @Override
+    public void clear() {}
+
+    @Override
+    public Index index() {
+        return index;
+    }
+
+    public static class ScriptLongLeafFieldData implements LeafNumericFieldData {
+        private final ScriptLongDocValues scriptBinaryDocValues;
+
+        ScriptLongLeafFieldData(ScriptLongDocValues scriptBinaryDocValues) {
+            this.scriptBinaryDocValues = scriptBinaryDocValues;
+        }
+
+        @Override
+        public ScriptDocValues<?> getScriptValues() {
+            return new ScriptDocValues.Longs(getLongValues());
+        }
+
+        @Override
+        public SortedBinaryDocValues getBytesValues() {
+            return FieldData.toString(scriptBinaryDocValues);
+        }
+
+        @Override
+        public SortedNumericDoubleValues getDoubleValues() {
+            return FieldData.castToDouble(getLongValues());
+        }
+
+        @Override
+        public SortedNumericDocValues getLongValues() {
+            return scriptBinaryDocValues;
+        }
+
+        @Override
+        public long ramBytesUsed() {
+            return 0;
+        }
+
+        @Override
+        public void close() {
+
+        }
+    }
+
+    static class ScriptBinaryResult {
```

Comment:
this is not needed right?

Reply:
needs to be removed also in the binary one, it's unused :)

Reply:
:+1:
I remember looking at it and thinking "this isn't used" and then forgetting.....